networkmanager/networkmanager.go

```diff
@@ -140,33 +143,42 @@ type outputAccessConfig struct {
 	Protocol string `json:"protocol"`
 }
 
-/*******************************************************************************
+/***********************************************************************************************************************
  * Vars
- ******************************************************************************/
+ **********************************************************************************************************************/
 
-var skipNetworkFileNames = []string{"lock", "last_reserved_ip.0"}
+var skipNetworkFileNames = []string{"lock", "last_reserved_ip.0"} // nolint:gochecknoglobals
 
-/*******************************************************************************
+var CNIConfListFromBytes func(bytes []byte) (*cni.NetworkConfigList, error) // nolint:gochecknoglobals
+
+/***********************************************************************************************************************
  * Public
- ******************************************************************************/
+ **********************************************************************************************************************/
 
-// New creates network manager instance
-func New(cfg *config.Config, trafficStorage TrafficStorage) (manager *NetworkManager, err error) {
+// New creates network manager instance.
+func New(cfg *config.Config, cniInterface cni.CNI, trafficStorage TrafficStorage) (manager *NetworkManager, err error) {
 	log.Debug("Create network manager")
 
 	cniDir := path.Join(cfg.WorkingDir, "cni")
 
 	manager = &NetworkManager{
-		hosts:      cfg.Hosts,
-		cniConfig:  cni.NewCNIConfigWithCacheDir([]string{cniBinPath}, cniDir, nil),
-		networkDir: path.Join(cniDir, "networks"),
+		hosts:             cfg.Hosts,
+		cniInterface:      cniInterface,
+		networkDir:        path.Join(cniDir, "networks"),
+		networkInstanceIP: make(map[string]map[string]string),
+	}
+
+	if manager.cniInterface == nil {
+		manager.cniInterface = cni.NewCNIConfigWithCacheDir([]string{cniBinPath}, cniDir, nil)
 	}
 
+	CNIConfListFromBytes = cni.ConfListFromBytes
```

Comment:
Set default value in place where it is defined:
```go
var CNIConfListFromBytes = cni.ConfListFromBytes
```

Reply:
Done